arkane/reference.py

```diff
@@ -121,10 +121,8 @@ def reference_data(self):
     def reference_data(self, value):
         if not value:
             self._reference_data = {}
-        elif isinstance(value, dict):
-            if all(isinstance(source, str) for source in value.keys()):
-                if all(isinstance(data_entry, ReferenceDataEntry) for data_entry in value.values()):
-                    self._reference_data = value
+        elif isinstance(value, dict) and _is_valid_reference_data(value):
+            self._reference_data = value
```

Comment:
I think I commented on this in an outdated version (commits aren't in order), but I think this change might fix my previous comment.